src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -387,42 +336,59 @@ bool add_node(
   return true;
 }
 
-void string_dependenciest::for_each_successor(
-  const nodet &node,
-  const std::function<void(const nodet &)> &f) const
+void string_dependenciest::for_each_dependency(
+  const builtin_function_nodet &node,
+  const std::function<void(const string_nodet &)> &f) const
 {
-  if(node.kind == nodet::BUILTIN)
+  for(const auto &s : node.data->string_arguments())
   {
-    const auto &builtin = builtin_function_nodes[node.index];
-    for(const auto &s : builtin->string_arguments())
+    std::vector<std::reference_wrapper<const exprt>> stack({s});
+    while(!stack.empty())
     {
-      std::vector<std::reference_wrapper<const exprt>> stack({s});
-      while(!stack.empty())
+      const auto current = stack.back();
+      stack.pop_back();
+      if(const auto if_expr = expr_try_dynamic_cast<if_exprt>(current.get()))
       {
-        const auto current = stack.back();
-        stack.pop_back();
-        if(const auto if_expr = expr_try_dynamic_cast<if_exprt>(current.get()))
-        {
-          stack.emplace_back(if_expr->true_case());
-          stack.emplace_back(if_expr->false_case());
-        }
-        else if(const auto node = node_at(to_array_string_expr(current)))
-          f(nodet(*node));
-        else
-          UNREACHABLE;
+        stack.emplace_back(if_expr->true_case());
+        stack.emplace_back(if_expr->false_case());
+      }
+      else if(const auto string_node = node_at(to_array_string_expr(current)))
+        f(*string_node);
+      else
+      {
+        std::cout << "Warning no node for " << format(current) << std::endl;
+        //UNREACHABLE;
```

Comment:
Why is this commented out?

Reply:
I reestablish it in a later commit that could be squashed with this one.
 https://github.com/diffblue/cbmc/pull/1947/commits/ec6321f7e2a0ffeab0d4071f4a2b9b94543af604